docs/guides/Custom_Native_Nodes.md

```diff
@@ -0,0 +1,133 @@
+# Custom Native Nodes
+
+One of Elementary's primary goals is to provide a comprehensive vocabulary of
+native audio processing nodes from which functional, declarative composition
+can derive a huge set of higher level audio processes. But sometimes that's not
+quite enough; sometimes you will need particular audio processing primitives
+that Elementary doesn't provide:
+
+* maybe you have existing DSP in C++ that you want to lean on
+* maybe the constraints around expressing feedback loops in JavaScript preclude the type of audio process that you want to build
+* maybe you have certain performance considerations for a particular, static piece of your process
+
+Whatever your reason, Elementary's engine offers an API for extending the vocabulary
```

Comment:
It's not clear which vocabulary is extended here. Perhaps this could be rewritten like this:

```suggestion
Whatever your reason, Elementary's engine offers an API for extending its vocabulary
```

It may not be the engine's vocabulary that is extended though, so maybe something else to make this more explicit.